syntax = "proto3";

package bluetooth.hci;

import "google/protobuf/empty.proto";
import "facade/common.proto";

service LeAclManagerFacade {
  rpc CreateConnection(bluetooth.facade.BluetoothAddressWithType) returns (stream LeConnectionEvent) {}
  rpc CancelConnection(bluetooth.facade.BluetoothAddressWithType) returns (google.protobuf.Empty) {}
  rpc Disconnect(LeHandleMsg) returns (google.protobuf.Empty) {}
  rpc SendAclData(LeAclData) returns (google.protobuf.Empty) {}
  rpc FetchAclData(google.protobuf.Empty) returns (stream LeAclData) {}
  rpc FetchIncomingConnection(google.protobuf.Empty) returns (stream LeConnectionEvent) {}
}

message LeHandleMsg {
  uint32 handle = 1;
}

message LeConnectionEvent {
  bytes event = 1;
}

message LeAclData {
  uint32 handle = 1;
  bytes payload = 2;
}

